src/patient-registration/patient-registration-validation.tsx

```diff
@@ -1,16 +1,35 @@
 import * as Yup from 'yup';
+import { useConfig } from '@openmrs/esm-module-config';
+import { AttributeValue } from './patient-registration-helper';
 
-export const validationSchema = Yup.object({
-  givenName: Yup.string().required('Given name is required'),
-  familyName: Yup.string().required('Family name is required'),
-  gender: Yup.string()
-    .oneOf(['Male', 'Female', 'Other', 'Unknown'], 'Gender is unspecified')
-    .required('Gender is required'),
-  birthdate: Yup.date()
-    .required('Birthdate is required')
-    .max(Date(), 'Birthdate cannot be in the future')
-    .nullable(),
-  yearsEstimated: Yup.number().min(0, 'Years cannot be less than 0'),
-  monthsEstimated: Yup.number().min(0, 'Months cannot be less than 0'),
-  telephoneNumber: Yup.string().matches(/^[0-9]*$/, 'Telephone number should only contain digits'),
-});
+export function validationSchema(props) {
+  return Yup.object({
+    givenName: Yup.string().required('Given name is required'),
+    familyName: Yup.string().required('Family name is required'),
+    gender: Yup.string()
+      .oneOf(['Male', 'Female', 'Other', 'Unknown'], 'Gender is unspecified')
+      .required('Gender is required'),
+    birthdate: Yup.date()
+      .required('Birthdate is required')
+      .max(Date(), 'Birthdate cannot be in the future')
+      .nullable(),
+    yearsEstimated: Yup.number().min(0, 'Years cannot be less than 0'),
+    monthsEstimated: Yup.number().min(0, 'Months cannot be less than 0'),
+    //telephoneNumber: Yup.string().matches(/^[0-9]*$/, 'Telephone number should only contain digits'),
+    attributes: Yup.array().of(
+      Yup.object().shape({
+        value: Yup.string().required('Required'),
+      }),
+    ),
+    /*  attributes: Yup.array(),
+    'attributes[0]': Yup.object().shape({
```

Comment:
@brandones any thoughts on how to make this validation work by accessing an object at a specific index of the array?

Reply:
This is used to validate the form input? Do we expect that the data will look something like

```javascript
attributes: [ { id: "foo", value: "something" }, ...]
```
?

I haven't used Yup before but based on [the docs](https://github.com/jquense/yup#yup) it looks like that isn't possible.

If you can format the data like

```javascript
attributes: { foo: { value: "something" }, ...]
```

then it seems like you should be able to do validation with Yup like

```javascript
attributes: Yup.object().shape(
  Object.fromEntries(
    config.personAttributes.map(attr =>
      [
        attr.id,
        Yup.object().shape({
          value: attr.validation.required ?
            Yup.string().required(`${attr.label} is required`) : 
            Yup.string()
        })
      ]
    )
  )
)
```

Hopefully that makes sense and works.

Reply:
Thanks @brandones I will look at what uv shared more deeply...Otherwise been tinkering around [after coming across this](https://github.com/jquense/yup/issues/528)

Reply:
Unfortunately, doing something like that would mess up the structure which we are using in the patient object...And so current there is a default on all of the attributes to be required

Reply:
So what's the plan? Do you have another idea?

You may end up needing to transform that structure, either on its way into validation or on submit or something.

Reply:
I figured out a way and implemented four validation rules...